google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSource.java

```diff
@@ -127,17 +115,17 @@ public File getFile() {
         new File(mavenProject.getBuildDirectory()).getPath() + File.separator
             + mavenProject.getFinalName();
 
-    XmlFile appEngineWebXml = AppEngineAssetProvider.getInstance()
-        .loadAppEngineStandardWebXml(project, Collections.singletonList(getModule()));
     AppEngineProjectService projectService = AppEngineProjectService.getInstance();
 
     // The environment will be null for newly deserialized deployment sources to ensure freshness.
     // In this case, we need to reload the environment.
     if (environment == null) {
-      environment = projectService.getModuleAppEngineEnvironment(appEngineWebXml);
+      environment = projectService.getModuleAppEngineEnvironment(getModule()).orElseThrow(
+          () -> new RuntimeException("No environment.")
+      );
     }
 
-    if (environment.isFlexible() && !projectService.isFlexCompat(appEngineWebXml)) {
+    if (environment.isFlexible()) {
```

Comment:
(Or, if you do like I just did, and add <env>flex</env> without opening the run config again and persist the new state, it will be APP_ENGINE_STANDARD. But never APP_ENGINE_FLEXIBLE)